iroh/src/protocol.rs

```diff
@@ -495,4 +497,56 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    async fn test_graceful_shutdown() -> TestResult {
+        #[derive(Debug, Clone, Default)]
+        struct TestProtocol {
+            connections: Arc<Mutex<Vec<Connection>>>,
+        }
+
+        const TEST_ALPN: &[u8] = b"/iroh/test/1";
+
+        impl ProtocolHandler for TestProtocol {
+            fn accept(&self, connection: Connection) -> BoxFuture<Result<()>> {
+                let this = self.clone();
+                Box::pin(async move {
+                    this.connections.lock().expect("poisoned").push(connection);
+                    Ok(())
+                })
+            }
+
+            fn shutdown(&self) -> BoxFuture<()> {
+                let this = self.clone();
+                Box::pin(async move {
+                    tokio::time::sleep(Duration::from_millis(100)).await;
+                    let mut connections = this.connections.lock().expect("poisoned");
+                    for conn in connections.drain(..) {
+                        conn.close(42u32.into(), b"shutdown");
+                    }
+                })
+            }
+        }
+
+        let endpoint = Endpoint::builder().bind().await?;
```

Comment:
I thought it might make sense for some tests to set `.relay_mode(RelayMode::Disabled)`, if they don't need it, since that should speed up getting a `node_addr` a bit.

Reply:
Changed that